### PR TITLE
add chromedriverPorts cap which allows specifying multiple ports or a range

### PR DESCRIPTION
**Pull request by jlipps**
this helps support the multiple webview case, and would provide a supported 'workaround' for e.g. https://github.com/appium/appium/issues/11820

### CODE REVIEW

**Comment by jlipps** [comment]
oh, also deprecate `chromeDriverPort` in favor of the more standard `chromedriverPort`, for parity with all our other chromedriver caps

**Comment by jlipps** [comment]
while i was in there i also removed a discrepancy where the bootstrapPort opt was only a CLI opt and not a desired cap. now it can be either like all the rest.

File: lib/commands/context.js [hunk 1]
```diff
@@ -265,16 +265,64 @@ async function setupExistingChromedriver (chromedriver) {
...
+    return port;
+  }
+
+  // otherwise find the free port based on a list or range provided by the user
+  // which could look like [8000, 8001, [9000, 9500]], where each element
+  // in the array is either a single port or an array with two elements
+  // defining a port range
+  log.debug(`Finding a free port for chromedriver using spec ${JSON.stringify(opts.chromedriverPorts)}`);
+  let foundPort = null;
+  for (let potentialPort of opts.chromedriverPorts) {
```
Comment (KazuCocoa) [inline lib/commands/context.js]: const?

File: lib/commands/context.js [hunk 2]
```diff
@@ -265,16 +265,64 @@ async function setupExistingChromedriver (chromedriver) {
   return chromedriver;
 }
 
+async function getChromedriverPort (opts) {
```
Comment (KazuCocoa) [inline lib/commands/context.js]: docstring?

File: lib/commands/context.js [hunk 3]
```diff
@@ -265,16 +265,64 @@ async function setupExistingChromedriver (chromedriver) {
...
+}
+
 helpers.setupNewChromedriver = async function setupNewChromedriver (opts, curDeviceId, adb) {
-  // if a port wasn't given, pick a random available one
-  if (!opts.chromeDriverPort) {
-    const getPort = B.promisify(PortFinder.getPort, {context: PortFinder});
-    opts.chromeDriverPort = await getPort();
-    log.debug(`A port was not given, using random port: ${opts.chromeDriverPort}`);
+  if (opts.chromeDriverPort) {
+    log.warn(`The 'chromeDriverPort' capability is deprecated. Please use 'chromedriverPort' instead`);
```
Comment (KazuCocoa) [inline lib/commands/context.js]: 👍

**Comment by jlipps** [comment]
ok, feedback addressed. this PR now also depends on https://github.com/appium/appium-base-driver/pull/314

**Comment by KazuCocoa** [comment]
a note need to update after this:

- https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md

**Comment by jlipps** [comment]
published in 4.14.0

**Pull request merged**